pkg/filter/stream/jwtauthn/matcher.go

```diff
@@ -0,0 +1,108 @@
+package jwtauthn
+
+import (
+	"net/url"
+	"strings"
+
+	envoy_config_route_v3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	routev3 "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	jwtauthnv3 "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/jwt_authn/v3"
+	"mosn.io/api"
+	"mosn.io/mosn/pkg/log"
+)
+
+// Matcher supports matching a HTTP requests with JWT requirements.
+type Matcher interface {
+	Matches(headers api.HeaderMap, requestPath string) bool
+}
+
+// NewMatcher creates a new Matcher.
+func NewMatcher(rule *jwtauthnv3.RequirementRule) Matcher {
+	switch rule.GetMatch().PathSpecifier.(type) {
+	case *envoy_config_route_v3.RouteMatch_Prefix:
+		return newPrefixMatcher(rule)
+
+	default:
+		// *envoy_config_route_v3.RouteMatch_Path
+		return newPathMatcher(rule)
+	}
+}
+
+type baseMatcher struct {
+	caseSensitive   bool
+	headers         []*routev3.HeaderMatcher
+	queryParameters []*routev3.QueryParameterMatcher
+}
+
+func newBaseMatcher(rule *jwtauthnv3.RequirementRule) *baseMatcher {
+	caseSensitive := true
+	if rule.Match.CaseSensitive != nil {
+		caseSensitive = rule.Match.CaseSensitive.Value
+	}
+
+	return &baseMatcher{
+		caseSensitive:   caseSensitive,
+		headers:         rule.GetMatch().GetHeaders(),
+		queryParameters: rule.GetMatch().GetQueryParameters(),
+	}
+}
+
+// Check match for HeaderMatcher and QueryParameterMatcher
+func (p *baseMatcher) matchRoutes(headers api.HeaderMap, requestPath string) bool {
+	matchs := matchHeaders(headers, p.headers)
+	if len(p.queryParameters) > 0 {
+
```

Comment:
?

Reply:
At present, only extracting token from headers is implemented, and extracting from query parameter has not been implemented yet